features/datachoices/src/main/java/org/opennms/features/datachoices/internal/UsageStatisticsReporter.java

```diff
@@ -180,9 +204,83 @@ public UsageStatisticsReportDTO generateReport() {
         usageStatisticsReport.setAlarms(m_alarmDao.countAll());
         // Node statistics
         usageStatisticsReport.setNodesBySysOid(m_nodeDao.getNumberOfNodesBySysOid());
-
+        // Location and Minion statistics
+        usageStatisticsReport.setMonitoredServices(m_monitoringLocationDao.countAll());
+        usageStatisticsReport.setMinions(m_minionDao.countAll());
+        // Time-series, sink, and RPC strategies
+        usageStatisticsReport.setTimeSeriesStrategy(TimeSeries.getTimeseriesStrategy().toString());
+        usageStatisticsReport.setSinkStrategy(SinkStrategy.getSinkStrategy().toString());
+        usageStatisticsReport.setRpcStrategy(RpcStrategy.getRpcStrategy().toString());
+        // User and group counts
+        try {
+            UserFactory.init();
+            usageStatisticsReport.setNumUsers(UserFactory.getInstance().getUserNames().size());
+            GroupFactory.init();
+            usageStatisticsReport.setNumGroups(GroupFactory.getInstance().getGroupNames().size());
+        } catch (IOException e) {
+            LOG.error("Encountered IOException while computing user and/or group counts. Setting both to -1.");
+            usageStatisticsReport.setNumUsers(-1);
+            usageStatisticsReport.setNumGroups(-1);
+        }
+        // Karaf features
+        usageStatisticsReport.setKarafFeatureList(computeKarafFeatureList());
         return usageStatisticsReport;
     }
+    
+    private List<String> computeKarafFeatureList() {
+        Set<String> featureSet = new HashSet<>();
+        File etcDir = new File(new File(System.getProperty("opennms.home")), "etc");
+        File featuresBootDotDDir = new File(etcDir, "featuresBoot.d");
+        File featuresCfgFile = new File(etcDir, "org.apache.karaf.features.cfg");
+        Properties featuresProp = new Properties();
+        try (FileInputStream fis = new FileInputStream(featuresCfgFile)) {
+            featuresProp.load(fis);
+            String bootProp = featuresProp.getProperty("featuresBoot");
+            if (bootProp != null) {
+                for (String propName : bootProp.split(",")) {
+                    if (propName.startsWith("(") && propName.length() > 1) {
+                        propName = propName.substring(1);
+                    }
+                    if (propName.endsWith(")") && propName.length() > 1) {
+                        propName = propName.substring(0, propName.length() - 2);
+                    }
+                    featureSet.add(propName.trim());
+                }
+            }
+        } catch (IOException e) {
+            LOG.error("Encountered IOException while loading Karaf features config. Features list will be empty.");
+        }
+        for (File featureDotDFile : featuresBootDotDDir.listFiles()) {
```

Comment:
Would be nice to leverage the existing logic in the `karaf-extender` instead of duplicating that here.

Reply:
Done in latest revisions. Thanks for the suggestion!

Reply:
Well, I thought I'd gotten it done and tested successfully, but now I'm getting an NPE in the for() loop. I know it's because `extender` is null. I thought that annotating the `m_extender` field with `@Reference` would get the job done. Afraid I've hit the limit of my Karaf-fu.

Reply:
Admin guide updated.